x/incentive/keeper/keeper_fees_test.go

```diff
@@ -94,3 +94,139 @@ func TestCollectGasFeesToIncentiveModule(t *testing.T) {
 	// It should be 9 usdc
 	require.Equal(t, collectedAmt, sdk.Coins{sdk.NewCoin(ptypes.USDC, sdk.NewInt(9))})
 }
+
+func TestCollectDEXRevenueToIncentiveModule(t *testing.T) {
+	app := simapp.InitElysTestApp(initChain)
+	ctx := app.BaseApp.NewContext(initChain, tmproto.Header{})
+
+	ik, bk, amm, oracle := app.IncentiveKeeper, app.BankKeeper, app.AmmKeeper, app.OracleKeeper
+
+	// Recalculate total committed info
+	ik.UpdateTotalCommitmentInfo(ctx)
+
+	// Setup coin prices
+	SetupStableCoinPrices(ctx, oracle)
+
+	// Generate 1 random account with 1000stake balanced
+	addr := simapp.AddTestAddrs(app, ctx, 2, sdk.NewInt(1000000))
+
+	// Create 2 pools
+
+	// #######################
+	// ####### POOL 1 ########
+	// Mint 100000USDC
+	usdcToken := sdk.NewCoins(sdk.NewCoin(ptypes.USDC, sdk.NewInt(100000)))
+
+	err := bk.MintCoins(ctx, types.ModuleName, usdcToken)
+	require.NoError(t, err)
+	err = bk.SendCoinsFromModuleToAccount(ctx, types.ModuleName, addr[0], usdcToken)
+	require.NoError(t, err)
+
+	var poolAssets []ammtypes.PoolAsset
+	// Elys
+	poolAssets = append(poolAssets, ammtypes.PoolAsset{
+		Weight: sdk.NewInt(50),
+		Token:  sdk.NewCoin(ptypes.Elys, sdk.NewInt(100000)),
+	})
+
+	// USDC
+	poolAssets = append(poolAssets, ammtypes.PoolAsset{
+		Weight: sdk.NewInt(50),
+		Token:  sdk.NewCoin(ptypes.USDC, sdk.NewInt(10000)),
+	})
```

Comment:
Can you initiate this poolAssets on a single operation without using append function?

Reply:
Okay

Reply:
Something like this would be recommended
```
poolAssets := []ammtypes.PoolAsset{
 {
		Weight: sdk.NewInt(50),
		Token:  sdk.NewCoin(ptypes.Elys, sdk.NewInt(100000)),
 },
 {
		Weight: sdk.NewInt(50),
		Token:  sdk.NewCoin(ptypes.USDC, sdk.NewInt(10000)),
 },
}
```

Reply:
Cool, awesome!